### ISSUE TITLE
(aws-ecs): Windows EC2 user data: clearing the ecs cache does not work anymore since AMI Windows_Server-2019-English-Core-ECS_Optimized-2022.05.12

### ISSUE THREAD

**Issue by mmarko08154711** [aws/aws-cdk#20711]
### Describe the bug

Starting with the latest AMI (Windows_Server-2019-English-Core-ECS_Optimized-2022.05.12, ECS tools v0.9.1) EC2 instances spawned by auto scaling will not connect to the cluster anymore.

The error in the instance startup:
`ECS Exec artifacts archive not found in cache: C:\ProgramData\Amazon\ECS\cache\execute-command-binaries.zip
At C:\Program Files\WindowsPowerShell\Modules\ECSTools\ECSTools.psm1:82 char:5`

Earlier AMIs using v0.9.0 work as expected.

From the AWS support:
`Recently introduced feature of exec for Windows containers [1] include pre-build exec binary within the "cache" folder hence the above line in user data to clean up cache causing issue which fails node to join cluster.

Our service team has been acknowledged on this and is currently working on a fix. The current recommendation from our backend team is to not cleanup ECS cache. Can you try removing that line and see if that help with the issue?`

The problematic CDK code (cluster.ts):
`
  private configureWindowsAutoScalingGroup(autoScalingGroup: autoscaling.AutoScalingGroup, options: AddAutoScalingGroupCapacityOptions = {}) {
    // clear the cache of the agent
    autoScalingGroup.addUserData('Remove-Item -Recurse C:\\ProgramData\\Amazon\\ECS\\Cache');
`

### Expected Behavior

EC2 instances connects to the cluster.

### Current Behavior

EC instances do not connect to the cluster.

### Reproduction Steps

n.a.

### CDK CLI Version

2.27.0 (build 8e89048)

### Node.js Version

v18.0.0

### OS

Windows 10

### Language

Typescript

**Comment by lenonestrella** [comment]
Having the same issue.

Is it possible to override the UserData content in the meanwhile?

**Comment by RexMorgan** [comment]
This is our work around, it seems to work.

```
const asg = new autoscaling.AutoScalingGroup(....);

const lines = (asg.userData as any).lines as string[];

// Blowing away this cache is problematic now
// https://github.com/aws/aws-cdk/issues/20711
delete lines[0];
```

**Comment by mmarko08154711** [comment]
Update: was fixed in the AWS AMI with ami-0b6c464dbdf96446d.
Leaving this ticket open because there has to be a proper way to circumvent problems with user data when something like this happens in the future.
